generators/javascript/lists.js

```diff
@@ -82,61 +82,63 @@ Blockly.JavaScript['lists_indexOf'] = function(block) {
 Blockly.JavaScript['lists_getIndex'] = function(block) {
   // Get element at index.
   // Note: Until January 2013 this block did not have MODE or WHERE inputs.
-  var mode = block.getFieldValue('MODE') || 'GET';
-  var where = block.getFieldValue('WHERE') || 'FROM_START';
-  var listOrder = (where === 'RANDOM') ? Blockly.JavaScript.ORDER_NONE :
+  const mode = block.getFieldValue('MODE') || 'GET';
+  const where = block.getFieldValue('WHERE') || 'FROM_START';
+  const listOrder = (where === 'RANDOM') ? Blockly.JavaScript.ORDER_NONE :
       Blockly.JavaScript.ORDER_MEMBER;
-  var list = Blockly.JavaScript.valueToCode(block, 'VALUE', listOrder) || '[]';
+  const list = Blockly.JavaScript.valueToCode(block, 'VALUE', listOrder) || '[]';
 
   switch (where) {
     case ('FIRST'):
       if (mode === 'GET') {
-        var code = list + '[0]';
+        const code = list + '[0]';
         return [code, Blockly.JavaScript.ORDER_MEMBER];
       } else if (mode === 'GET_REMOVE') {
-        var code = list + '.shift()';
+        const code = list + '.shift()';
         return [code, Blockly.JavaScript.ORDER_MEMBER];
       } else if (mode === 'REMOVE') {
         return list + '.shift();\n';
       }
       break;
     case ('LAST'):
       if (mode === 'GET') {
-        var code = list + '.slice(-1)[0]';
+        const code = list + '.slice(-1)[0]';
         return [code, Blockly.JavaScript.ORDER_MEMBER];
       } else if (mode === 'GET_REMOVE') {
-        var code = list + '.pop()';
+        const code = list + '.pop()';
         return [code, Blockly.JavaScript.ORDER_MEMBER];
       } else if (mode === 'REMOVE') {
         return list + '.pop();\n';
       }
       break;
-    case ('FROM_START'):
-      var at = Blockly.JavaScript.getAdjusted(block, 'AT');
+    case ('FROM_START'): {
```

Comment:
Below on line 277 you solve the same problem with `let at1;` instead of creating extra scopes.  Is there a reason not to be consistent?

Reply:
Yes. If the variable will be used outside of that scope, I create it before the switch (with `let`). If it will only be used inside the case, I add brackets to create a new scope.